front-end-tools/src/components/InitComponent.tsx

```diff
@@ -0,0 +1,643 @@
+                                    ? moduleReferenceDeployed
+                                    : moduleReferenceCalculated;
+
+                            if (checkboxElement && newModuleReference !== undefined) {
+                                form.setValue('moduleReference', newModuleReference);
+
+                                setDisplayContracts(contracts);
+                            }
+                        }}
+                    />
+                </Form.Group>
+
+                {useModuleReferenceFromStep1 && (
+                    <>
+                        <br />
+                        <Alert variant="info">
+                            <div>
+                                This checkbox autofilled the <code>module reference</code>, the{' '}
+                                <code>smart contract name</code>, and the <code>input parameter schema</code> from the
+                                module in step1.
+                            </div>
+                            <br />
+                            <div>
+                                <b>Uncheck</b> this box, if you want to manually fill in a <code>module reference</code>
+                                , the <code>smart contract name</code>, or an <code>input parameter schema</code>.
+                            </div>
+                            <br />
+                            <div>
+                                <b>Uncheck</b> and <b>check</b> this box again, if you want to load a new module from
+                                step 1.
+                            </div>
+                        </Alert>
+                        <br />
+                    </>
+                )}
+
+                {moduleReferenceError && <Alert variant="danger"> Error: {moduleReferenceError}. </Alert>}
+
+                <Row>
+                    <Form.Group className="col-md-4 mb-3">
+                        <Form.Label> Module Reference</Form.Label>
+                        <Form.Control
+                            defaultValue="91225f9538ac2903466cc4ab07b6eb607a2cd349549f357dfdf4e6042dde0693"
+                            disabled={!!useModuleReferenceFromStep1}
+                            {...form.register('moduleReference', { required: true })}
+                            onChange={(e) => {
+                                const register = form.register('moduleReference', {
+                                    required: true,
+                                });
+
+                                register.onChange(e);
+
+                                setModuleReferenceLengthError(undefined);
+
+                                const moduleRef = form.getValues('moduleReference');
+
+                                if (moduleRef !== undefined && moduleRef.length !== 64) {
+                                    setModuleReferenceLengthError('Module reference has to be of length 64');
+                                }
```

Comment:
I experimented with it but the problem is that we want to give feedback to the user as early as possible for some of the critical input fields (input values that are needed for e.g. loading additional information such as schemas). That is why some custom evaluations and more precise error reporting are done here to guide the user to resolve the issue.

If we wait until the user presses the `submitForm` button or give a very general error here (input value does not match defined pattern) seems not so helpful to the user.

Reply:
It might make sense to validate instantly without waiting for submission for these forms, but the library also supports that: https://react-hook-form.com/docs/useform#mode

```ts
const form = useForm({mode: 'all'});
```

Reply:
I added the `{mode: 'all'}` to all forms.